ocdskingfisherarchive/scrapy_log_file.py

```diff
@@ -6,13 +7,79 @@
 class ScrapyLogFile():
 
     def __init__(self, file_name):
-        with open(file_name) as fp:
+        self._file_name = file_name
+        # We lazy-parse any data we need
+        # This may be called on a directory of very big logs and we want it to be efficient.
+        # None means not parsed
+        self._logparser_data = None
+        self._errors_sent_to_process_count = None
+        self._spider_arguments = None
+
+    # Logparser Processing
+
+    def _process_logparser(self):
+        with open(self._file_name) as fp:
             text = fp.read()
         # If you pass headlines=0, taillines=0 the returned dict has all the raw lag data in, which may be very large
         # So we request 1 line for each even tho we don't need it
-        self.log_data = parse(text, headlines=1, taillines=1)
+        self._logparser_data = parse(text, headlines=1, taillines=1)
 
     def does_match_date_version(self, date_version):
-        start_time = datetime.datetime.strptime(self.log_data['first_log_time'], '%Y-%m-%d %H:%M:%S')
+        if self._logparser_data is None:
+            self._process_logparser()
+        start_time = datetime.datetime.strptime(self._logparser_data['first_log_time'], '%Y-%m-%d %H:%M:%S')
         diff = abs(start_time.timestamp() - date_version.timestamp())
         return diff < 3
+
+    # Manual Processing
+
+    def _process_manually(self):
```

Comment:
Why is this called `_process_manually`? Seems to be automated!

Since it's a long method, a docstring to say what it's doing would be good (counting the number or errors and extracting the spider arguments, from a quick read).

Reply:
I called it manual as opposed to by the library, but I can see that is confusing. Done.